api/handlers.go

```diff
@@ -486,6 +486,10 @@ func getContainersJSON(c *context, w http.ResponseWriter, r *http.Request) {
 // GET /containers/{name:.*}/json
 func getContainerJSON(c *context, w http.ResponseWriter, r *http.Request) {
 	name := mux.Vars(r)["name"]
+	sizeQuery := ""
+	if boolValue(r, "size") {
+		sizeQuery = "?size=1"
```

Comment:
This fixes `docker inspect --size <container_name>` so it works properly via swarm.